File: image.c
#include <SDL.h>
#include <SDL_image.h>
#include <stdbool.h>

#undef main

int main(void) {
  SDL_Init(SDL_INIT_EVERYTHING);
  IMG_Init(IMG_INIT_PNG | IMG_INIT_JPG);

  SDL_Window* win = SDL_CreateWindow(
    "Image example",
    SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
    400, 400,
    SDL_WINDOW_SHOWN
  );

  SDL_Renderer* render = SDL_CreateRenderer(win, -1, 0);
  SDL_Event ev;

  SDL_Surface* img = IMG_Load("rei_chikita.jpeg");
  SDL_Texture* texture = SDL_CreateTextureFromSurface(render, img);

  SDL_FreeSurface(img);

  if (win == NULL) {
    printf("Cannot create the window!\n");
    printf("Error: %s\n", SDL_GetError());
    return 1;
  }

  bool quit = false;

  while (!quit) {
    while (SDL_PollEvent(&ev) > 0) {
      if (ev.type == SDL_QUIT) quit = true;
    }

    SDL_RenderCopy(render, texture, NULL, NULL);
    SDL_RenderPresent(render);
  }

  SDL_DestroyTexture(texture);
  SDL_DestroyRenderer(render);
  SDL_DestroyWindow(win);

  IMG_Quit();
  SDL_Quit();

  return 0;
}

File: image.nim
import sdl2
import sdl2/image as img
import std/with

discard sdl2.init(INIT_EVERYTHING)
discard img.init()

var
  ev: Event
  win: WindowPtr
  render: RendererPtr
  surface: SurfacePtr
  texture: TexturePtr
  running = true

win = createWindow(
  "Image example",
  SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
  400, 400,
  SDL_WINDOW_SHOWN
)

render  = win.createRenderer(-1, 0)
surface = img.load("rei_chikita.jpeg")
texture = render.createTextureFromSurface(surface)

surface.freeSurface()

while running:
  while ev.pollEvent():
    if ev.kind == QuitEvent: running = false

  with render:
    copy(texture, nil, nil)
    present()

texture.destroy()
render.destroy()
win.destroy()

img.quit()
sdl2.quit()